distributed/scheduler.py

```diff
@@ -922,76 +1239,72 @@ def remove_worker(self, comm=None, address=None, safe=False, close=True):
         with log_errors():
             if self.status == 'closed':
                 return
-            if address not in self.processing:
+            if address not in self.workers:
                 return 'already-removed'
 
             address = self.coerce_address(address)
             host = get_address_host(address)
 
+            ws = self.workers[address]
+
             self.log_event(['all', address], {'action': 'remove-worker',
                                               'worker': address,
-                                              'processing-tasks': self.processing[address]})
+                                              'processing-tasks': ws.processing})
```

Comment:
It occurs to me that we should make a copy of the dict here (unrelated to this PR though).